tests/green_mode/green_modeTest.py

```diff
@@ -356,5 +374,150 @@ def test_bear_test_fun_2(self):
+                                              'filename': ['d.py']}]]}
+        self.assertEqual(data_struct, test_data_struct)
+        project_files = ['a.py', 'b.py', 'c.py', 'd.py']
+        coafile = '.coafile.green'
+        full_path = str(Path(__file__).parent.parent.parent)
+        full_path_coafile = str(Path(__file__).parent.parent.parent / coafile)
+        with patch('os.walk') as mockwalk:
+            mockwalk.return_val = mockwalk.return_value = [
+                ('', (), ('a.py', 'b.py', 'c.py', 'd.py')), ]
+            generate_green_mode_sections(data_struct, full_path,
+                                         project_files, ['x'],
+                                         printer)
+        contents = ""
+
+        with open(full_path_coafile) as f:
+            for line in f.readlines():
+                contents += line
+        full_path_glob = escape(full_path + os.sep + '**', '\\')
+        test_contents = dedent("""
+            [all]
+            ignore = x
+            [all.TestLocalBear1]
+            ignore += a.py, b.py, c.py, d.py
+            bears = TestLocalBear
+            some_setting = 3
+            files = a.py, b.py, {full_path_glob}
+            [all.TestLocalBear2]
+            ignore += a.py, b.py, c.py, d.py
+            files = c.py, {full_path_glob}
+            bears = TestLocalBear
+            some_setting = 4
+            [all.TestLocalBear3]
+            ignore += a.py, b.py, c.py, d.py
+            files = d.py, {full_path_glob}
+            bears = TestLocalBear
+            some_other_setting = x""").format(
+                full_path_glob=full_path_glob)
+        # Since the order of settings within a seciton is volatile.
+        print('test_contents')
+        for line in test_contents.split('\n'):
+            if line == 'ignore = x':
+                continue  # Since the path depends on the test directory
+                self.assertIn(line, [i.strip('\\').replace('\\\\C', 'C')
+                                     for i in contents.split('\n')])
+
+        with patch('os.walk') as mockwalk:
+            mockwalk.return_val = mockwalk.return_value = [
+                ('', (), ('a.py', 'b.py', 'c.py', 'd.py')), ]
+            generate_green_mode_sections(data_struct, full_path,
+                                         project_files, [],
+                                         printer)
+        contents = ""
+
+        with open(full_path_coafile) as f:
+            for line in f.readlines():
+                contents += line
+        # TODO: remove the prefix 'all.' from section names when section
+        # all is not present which only happends when the ignore field
+        # is empty.
+        test_contents = dedent("""
+            [all.TestLocalBear1]
+            ignore = a.py, b.py, c.py, d.py
+            files = a.py, b.py, {full_path_glob}
+            bears = TestLocalBear
+            some_setting = 3
+            [all.TestLocalBear2]
```

Comment:
add blank lines between sections.  Readability is important to verifying test cases are doing what they are supposed to do